cmd/expired-authz-purger/main.go

```diff
@@ -45,19 +46,38 @@ type expiredAuthzPurger struct {
 	batchSize int64
 }
 
-func (p *expiredAuthzPurger) purgeAuthzs(purgeBefore time.Time, yes bool) (int64, error) {
-	if !yes {
-		var count int
-		err := p.db.SelectOne(&count, `SELECT COUNT(1) FROM pendingAuthorizations AS pa WHERE expires <= ?`, purgeBefore)
-		if err != nil {
-			return 0, err
+func (p *expiredAuthzPurger) purge(table string, yes bool, purgeBefore time.Time) error {
+	var ids []string
+	for {
+		var idBatch []string
+		_, err := p.db.Select(
+			&idBatch,
+			fmt.Sprintf("SELECT id FROM %s WHERE expires <= ? LIMIT ? OFFSET ?", table),
```

Comment:
Using %s to interpolate into a SQL string is a code smell, even though it's safe here. Instead, what about:

```
var query string
switch table {
  case "pendingAuthorizations":
    query = "SELECT id FROM pendingAuthorizations WHERE expires <= ? LIMIT ? OFFSET ?"
  case "authz":
    query = "SELECT id FROM authz WHERE expires <= ? LIMIT ? OFFSET ?"
  default:
    return fmt.Errorf(...)
}
```